rector.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+use Rector\Config\RectorConfig;
+
+return RectorConfig::configure()
+    ->withPaths([
+        __DIR__ . '/protected/humhub',
+    ])
+    ->withSkip([
+        \Rector\Php81\Rector\Array_\FirstClassCallableRector::class,
+        \Rector\TypeDeclaration\Rector\ClassMethod\ReturnNeverTypeRector::class,
+        __DIR__ . '/protected/humhub/config/*',
+        __DIR__ . '/protected/humhub/messages/*',
+        __DIR__ . '/protected/humhub/modules/*/messages/*',
+        __DIR__ . '/protected/humhub/modules/*/config.php',
+    ])
+    ->withPhpSets(php82:true)
```

Comment:
@luke- shouldn't we specify that we want to use PHP 8.2 rules as base and include up to PHP 8.4 rules?
```php
    ->withPhpSets(
        php82: true,
        php83: true,
        php84: true
    )
 ```

Reply:
@marc-farre As I understand it, Rector migrates the code to PHP MinVersion 8.2 with `php82`.
When using `php84` it would already use PHP 8.4 features, which may not compatible with PHP 8.2.

Reply:
@luke- Yes, sorry, I got caught by AI mistake! I should have read the doc before...

The idea behind was to make sure that the code will also work with 8.3 and 8.4, but I was wrong.

However, the official doc https://getrector.com/documentation/set-lists#content-php-sets says:

The best practise is to use PHP version defined in composer.json. Rector will automatically pick it up with empty ->withPhpSets() method.

Perhaps we should go for that. This way, when we change PHP version, we don't forget to update rector.php too.
What do you think?

Reply:
@marc-farre good point! I've changed that in `develop`!